lib/srv/alpnproxy/proxy.go

```diff
@@ -193,7 +193,8 @@ type Proxy struct {
 	cfg                ProxyConfig
 	supportedProtocols []common.Protocol
 	log                logrus.FieldLogger
-	cancel             context.CancelFunc
+	cancelFunc         context.CancelFunc
+	mu                 sync.Mutex
```

Comment:
Suggestion: `mu` before guarded field, explicitly state what is guarded:

```suggestion
	// mu guards cancelFunc.
	mu                 sync.Mutex
	cancelFunc         context.CancelFunc
```